File: 05-retos/binary-search.js
search(nums, target) {
    let left = 0;
    let right = nums.length -1;
    while(left <= right){
        const mid = left + Math.floor((right- left) / 2);
        if (nums[mid] === target){
            return mid;
        }
        if (nums[mid]< target){
            left = mid + 1;
        } else {
            right = mid -1;
        }
    }
    return -1;
}

File: 05-retos/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        def binary_search(left:int, right:int):
            if left > right:
                return -1
            mid = (left+right)//2
            if nums[mid] == target:
                return mid
            if target < nums[mid]:
                return binary_search(left, mid-1)
            else:
                return binary_search(mid+1, right)
        return binary_search(0, len(nums)-1)
